src/clj/game/cards.clj

```diff
@@ -207,6 +207,69 @@
                                                (vals selected-cards)))]
                            (effect-completed state side (make-result eid {:number counter-count :msg msg})))))}))
 
+(defn pick-credit-providing-cards
+  "Similar to pick-virus-counters-to-spend. Works on :recurring and normal credits."
+  ([provider-func] (pick-credit-providing-cards provider-func (hash-map) 0 nil))
+  ([provider-func target-count] (pick-credit-providing-cards provider-func (hash-map) 0 target-count))
+  ([provider-func selected-cards counter-count target-count]
+   (let [pay-rest (req (if (pos? target-count)
+                         (do
+                           (deduct state side [:credit (- target-count counter-count)])
+                           (swap! state update-in [:stats side :spent :credit] (fnil + 0) target-count)
+                           (let [msg (str (join ", " (map #(let [{:keys [card number]} %
+                                                                 title (:title card)]
+                                                             (str number "[Credits] from " title))
+                                                          (vals selected-cards)))
+                                          (let [remainder (- target-count counter-count)]
+                                            (when (pos? remainder) (str " and " remainder "[Credits]"))))]
+                             (effect-completed state side (make-result eid {:number counter-count :msg msg}))))
+                         (effect-completed state side 0)))]
+     (if (or (not (pos? target-count))        ; there is a limit
+             (>= counter-count target-count)  ; paid everything
+             (zero? (count (provider-func)))) ; no more additional credit sources needed
+
+       {:effect pay-rest}
+
+       {:async true
+        :prompt (str "Select a credit providing card ("
+                     counter-count (when (and target-count (pos? target-count))
+                                     (str " of " target-count))
+                     " credits)")
+        :choices {:req #(and (in-coll? (map :cid (provider-func)) (:cid %))
+                             (case (-> % card-def :interactions :pay-credits :type)
+                               :recurring
+                               (pos? (get-counters % :recurring))
+
+                               :credit
+                               (pos? (get-counters % :credit))
+
+                               :custom
+                               true))}
+        :effect (req (let [pay-credits-type (-> target card-def :interactions :pay-credits :type)
+                           gained-credits (case pay-credits-type
+                                            :recurring
+                                            (or (add-prop state side target :rec-counter -1) 1)
```

Comment:
Is the goal to return a `1`? If so, I'd recommend using `do`, cuz `or` makes it look like you expect `add-prop` to fail or return incorrectly.

Reply:
Thanks! That was exactly what I wanted to do.